contracts/DVD/DVDTransferManager.sol

```diff
@@ -314,20 +330,25 @@ contract DVDTransferManager is Ownable {
         Delivery memory token2 = token2ToDeliver[_transferID];
         require(
             token1.counterpart != address(0) && token2.counterpart != address(0)
-            , "transfer ID does not exist");
+            , TransferIDDoesNotExist());
         IERC20 token1Contract = IERC20(token1.token);
         IERC20 token2Contract = IERC20(token2.token);
         require (
             msg.sender == token2.counterpart ||
             isTREXAgent(token1.token, msg.sender) ||
             isTREXAgent(token2.token, msg.sender)
-            , "transfer has to be done by the counterpart or by owner");
+            , TransferOnlyByCounterpartOrOwner());
+
+        uint256 balance = token2Contract.balanceOf(token2.counterpart);
         require(
-            token2Contract.balanceOf(token2.counterpart) >= token2.amount
-            , "Not enough tokens in balance");
+            balance >= token2.amount
+            , CommonErrors.ERC20InsufficientBalance(token2.counterpart, balance, token2.amount));
+
+        uint256 allowance = token2Contract.allowance(token2.counterpart, address(this));
         require(
-            token2Contract.allowance(token2.counterpart, address(this)) >= token2.amount
-            , "not enough allowance to transfer");
+            allowance >= token2.amount
+            , CommonErrors.ERC20InsufficientAllowance(token2.counterpart, allowance, token2.amount));
```

Comment:
not sure this one is necessary, as the allowance is checked by the transferFrom a few lines under